zanata-war/src/main/java/org/zanata/dao/ProjectDAO.java

```diff
@@ -255,15 +255,35 @@ public int getTotalObsoleteProjectCount() {
     public List<HProject> searchProjects(@Nonnull String searchQuery,
             int maxResult, int firstResult, boolean includeObsolete)
             throws ParseException {
-        FullTextQuery query = getTextQuery(searchQuery, includeObsolete);
-        return query.setMaxResults(maxResult).setFirstResult(firstResult)
-                .getResultList();
+
+        Query query = getSearchProjectQuery(searchQuery, includeObsolete);
+        query.setMaxResults(maxResult);
+        query.setFirstResult(firstResult);
+        query.setComment("ProjectDAO.searchProjects");
+        return query.list();
     }
 
     public int getQueryProjectSize(@Nonnull String searchQuery,
             boolean includeObsolete) throws ParseException {
-        FullTextQuery query = getTextQuery(searchQuery, includeObsolete);
-        return query.getResultSize();
+        Query query = getSearchProjectQuery(searchQuery, includeObsolete);
+        query.setComment("ProjectDAO.getQueryProjectSize");
+        return query.list().size();
+    }
+
+    private Query getSearchProjectQuery(@Nonnull String searchQuery,
+            boolean includeObsolete) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("from HProject p where ");
+
+        if(!includeObsolete) {
+            sb.append("p.status <> ")
+                .append("'" + EntityStatus.OBSOLETE.getInitial() + "' and ");
+        }
+        sb.append("lower(p.name) like :query or lower(p.slug) like :query");
```

Comment:
I think you need to put brackets around "lower(p.name) like :query or lower(p.slug) like :query". Even better put it as a named query in HProject since includeObsolete is almost always false now.